mm/slab.c

```diff
@@ -3315,7 +3314,7 @@ slab_alloc(struct kmem_cache *cachep, gfp_t flags, size_t orig_size, unsigned lo
 
 	objp = kfence_alloc(cachep, orig_size, flags);
 	if (objp)
-		goto leave;
+		return objp;
```

Comment:
Similar here. I'm worried about the pre- and post-alloc hooks. We need to understand if we need them.